dist/index.html

```diff
@@ -39,39 +39,45 @@
+                document.getElementById('id-container').className = 'unsupported';
 
-    } else {
-        // var Hoot = new Hoot();
-        var id = iD.Context();
+            } else {
+                var id = iD.Context();
 
-        // disable boundaries (unless we have an explicit disable_features list)
-        var q = iD.utilStringQs(window.location.hash.substring(1));
-        if (!q.hasOwnProperty('disable_features')) {
-            id.features().disable('boundaries');
-        }
+                // disable boundaries (unless we have an explicit disable_features list)
+                var q = iD.utilStringQs(window.location.hash.substring(1));
+                if (!q.hasOwnProperty('disable_features')) {
+                    id.features().disable('boundaries');
+                }
 
-        id.ui()(document.getElementById('id-sink'), function() {
-            Hoot.init( id );
-            id.container().select('#about-list')
-                .insert('li', '.user-list')
-                .attr('class', 'source-switch')
-                .call(iD.uiSourceSwitch(id)
-                    .keys([{
-                        'urlroot': 'https://www.openstreetmap.org',
-                        'oauth_consumer_key': '5A043yRSEugj4DJ5TljuapfnrflWDte8jTOcWLlT',
-                        'oauth_secret': 'aB3jKq1TRsCOUrfOIZ6oQMEDmv2ptV76PA54NGLL'
-                    }, {
-                        'urlroot': 'https://api06.dev.openstreetmap.org',
-                        'oauth_consumer_key': 'zwQZFivccHkLs3a8Rq5CoS412fE5aPCXDw9DZj7R',
-                        'oauth_secret': 'aMnOOCwExO2XYtRVWJ1bI9QOdqh1cay2UgpbhA6p'
-                    }
-                    ])
-                );
-        });
-    }
+                id.ui()(document.getElementById('id-sink'), function() {
+                    Hoot.init( id );
+                    id.container().select('#about-list')
+                        .insert('li', '.user-list')
+                        .attr('class', 'source-switch')
+                        .call(iD.uiSourceSwitch(id)
+                            .keys([{
+                                'urlroot': 'https://www.openstreetmap.org',
+                                'oauth_consumer_key': '5A043yRSEugj4DJ5TljuapfnrflWDte8jTOcWLlT',
+                                'oauth_secret': 'aB3jKq1TRsCOUrfOIZ6oQMEDmv2ptV76PA54NGLL'
+                            }, {
+                                'urlroot': 'https://api06.dev.openstreetmap.org',
+                                'oauth_consumer_key': 'zwQZFivccHkLs3a8Rq5CoS412fE5aPCXDw9DZj7R',
+                                'oauth_secret': 'aMnOOCwExO2XYtRVWJ1bI9QOdqh1cay2UgpbhA6p'
+                            }
+                            ])
+                        );
+                });
+            }
+        } )
+        .catch( () => { // user not logged in - redirect to login page
+            window.location.replace( '/login.html' );
```

Comment:
Is this going to work when the UI is not at the root, but under `/hootenanny-id` when deployed with tomcat?

Reply:
good catch. i'll update it so that only the last part of the path changes